octoprint_mrbeam/iobeam/dust_manager.py

```diff
@@ -203,15 +206,39 @@ def _send_fan_command(self, command, wait_time=-1.0, max_retries=-1):
 
 		return False
 
-	def _write_analytics(self, dust_start, dust_start_ts, dust_end, dust_end_ts):
-		self._logger.debug("dust extraction time {} from {} to {} (gradient: {})".format(dust_end_ts - dust_start_ts, dust_start, dust_end, (dust_start - dust_end) / (dust_end_ts - dust_start_ts)))
-		data = dict(
-			dust_start=dust_start,
-			dust_end=dust_end,
-			dust_start_ts=dust_start_ts,
-			dust_end_ts=dust_end_ts
-		)
-		_mrbeam_plugin_implementation._analytics_handler.add_dust_log(data)
+	def _send_dust_to_analytics(self,val):
+		"""
+		Sends dust value periodically to analytics_handler to get overall stats and dust profile.
+		:param val: measured dust value
+		:return:
+		"""
+		_mrbeam_plugin_implementation._analytics_handler.add_dust_value(val)
+
+	def _send_final_dust_to_analytics(self, dust_start, dust_start_ts, dust_end, dust_end_ts):
```

Comment:
This method should not be here but in the analytics package. (Eg. analytics_handler). 
All this method does is data preparation/conversion for analytics. Therefore it should be in the analytics package. Also AnalyticsKeys (as ak) should only be used WITHIN the analytics package IMO.
(I know, the capsulation in/around the iobeam package here is not really textbook style...)

Reply:
done